### PR TITLE
dynamic collection name

### PR DESCRIPTION
**Pull request by 0xb4lamx**
add the possibility to choose the collection name instead of "casbin_rule".

### CODE REVIEW

**Comment by hsluoyz** [comment]
@closetool please review.

File: adapter.go [hunk 1]
```diff
@@ -79,15 +82,15 @@ func NewAdapter(url string, timeout ...interface{}) (persist.BatchAdapter, error
...
 	}
 
-	return NewAdapterWithClientOption(clientOption, databaseName, timeout...)
+	return NewAdapterWithClientOption(clientOption, databaseName, defaultCollectionName, timeout...)
 }
 
 // NewAdapterWithClientOption is an alternative constructor for Adapter
 // that does the same as NewAdapter, but uses mongo.ClientOption instead of a Mongo URL
-func NewAdapterWithClientOption(clientOption *options.ClientOptions, databaseName string, timeout ...interface{}) (persist.BatchAdapter, error) {
+func NewAdapterWithClientOption(clientOption *options.ClientOptions, databaseName string, collectionName string, timeout ...interface{}) (persist.BatchAdapter, error) {
```
Comment (hsluoyz) [inline adapter.go:93]: Don't make breaking change. You can add new constructor for your new parameter.

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: adapter.go (hunk 1 above)
Comment (0xb4lamx) [reply adapter.go]: fixed.

the change of the default database name to `casbin` not `casbin_rule` might create an issue as well?
(although the db name must be "casbin" not "casbin_rule" as mentioned in the comment above )

Comment (hsluoyz) [reply adapter.go]: @0xb4lamx I think the DB name change is also a breaking change. So we have to make a major version release.

Comment (hsluoyz) [reply adapter.go]: Can you also add the 2 new constructors to our README?

Comment (0xb4lamx) [reply adapter.go]: done. ready to merge.

**Review by hsluoyz — APPROVED** [review]

**Comment by hsluoyz** [comment]
@0xb4lamx released: https://github.com/casbin/mongodb-adapter/releases/tag/v3.2.0

**Pull request merged**